lighthouse/service-selection/multi-cluster-service-selection.md

```diff
@@ -0,0 +1,513 @@
+        [3](https://dl.acm.org/doi/pdf/10.1145/346000.346004).
+
+1. User Location Distribution - should be picked at random
+   from a normal distribution of the given user count
+   [Twiter User Goe Distribution](https://www.researchgate.net/publication/315074173_Strategies_for_combining_Twitter_users_geo-location_methods)
+
+1. Topology:
+
+    1. Clusters Distribution:
+
+        1. Cloud Provider
+        1. Regions/Zones
+        1. Applications:
+
+    1. Clusters Layout:
+
+        1. Standard web apps (API based)
+        1. CDN and Gaming apps, high data volume
+        1. Data pipeline apps
+
+We shall compare different applications to the best of our knowledge.
+And would like to invite you and suggest layouts for different  applications.
+The rest of the testing parameters (load/capacity/random
+latency) will be determined using randomize tools.
+
+## Open questions
+
+- How often do deployments affect changes in the application topology?
+    1. How often deployment changes service placement
+    1. How will it affect the optimization?
+- How adaptive would we like to be for changes in the topology?
+    1. We might need to trigger data collection and optimization after a new deployment.
+- What data/metrics should we share through the Broker?
+- What should we collect from the local cluster perspective?
+    1. Pricing will have to be shared because they depend on the location/provider.
+    1. Latency and RTT and Load should be collected from a local perspective.
+- Cost Parameters/Metrics
+    1. How often would we like to collect the parameters?
+    1. Would we like to make the collection configurable?
+- How can we understand what cloud provider the cluster is on.
+- What cloud providers do we support?
+- Real-life testbed
+- More application topologies
+
+## Implementation History
+
+Currently, the Lighthouse DNS plugin within Submariner supports Round-Robin load
+balancing between Exported/Imported service in the multi-cluster deployment.
+
+## Future
+
+### Extensions
+
+1. Cost
+
+    1. Parameters -
+       we might want to think of more cost parameters to improve our model
+
+        1. Network congestion
+        1. Turnaround time - is the total time that it takes
+           between the submission of a request and the return of
+           the complete response to the customer. This is the
+           main metric for users experience we would like to
+           optimize it, as it derived from many of the variables
+           we mentioned above (latency/RTT/Application) on one
+           hand, but from the other, we are not sure yet, on how
+           to fit it in our one-hop model, we would tackle this
+           parameter in the future.
+
+    1. Functions - we might want to consider and test different
+       functions for combining the different parameters in the function
+
+### Challenges
+
+- Scalability to 5G – thousands of clusters
+- Autoscale utilization
+- Multi-hop

```

Comment:
Handling sessions and DNS caching may create challenges as well.

Reply:
True! Thanks again - will update